doc/input_system.md

```diff
@@ -0,0 +1,226 @@
+The input system
+================
+
+This document describes the custom input system in Thrive.
+The system uses C# Attributes attached to methods to call the correct code when the user does
+an input action.
+
+How to use attributes
+---------------------
+
+To apply an attribute to a method, you need to add a code line like this before a method: 
+```csharp
+[OneOfTheInputAttributes(parameters)]
+```
+
+The different available attributes and their parameters are described below.
+
+
+You can read more about C# attributes 
+[here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/).
+
+The available input attributes
+------------------------------
+
+| Attribute | Description | Parameters | Method parameters | Multiple |
+| --------- | ----------- | ---------- | ----------------- | -------- |
+| RunOnKey  | Fires repeatedly when the input is pressed | input : string | delta : float | yes |
+| RunOnKeyChange | Fires once when the input is pressed or released | input : string | none | yes |
+| RunOnKeyDown | Fires once when the input is pressed | input : string | none | yes |
+| RunOnKeyUp | Fires once when the input is released | input : string | none | yes |
+| RunOnKeyToggle | Fires once when the input is pressed | input : string | state : bool | yes |
```

Comment:
Shouldn't this be "Fires when the state of the toggle changes"
Because just above "pressed" seems to mean that the key is pressed down, so this is misleading documentation.